python/revng/cli/download_pdbs.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import sys
+
+import pefile
+import wget
+
+from .commands_registry import Command, Options, commands_registry
+from .support import log_error
+
+
+class DownloadPDBsCommand(Command):
+    def __init__(self):
+        super().__init__(("model", "download-pdbs"), "Download missing PDBs for CRTs.")
+
+    def register_arguments(self, parser):
+        parser.add_argument("input", type=str, default="", help="The input file.")
+        parser.add_argument(
+            "libpath",
+            type=str,
+            default="",
+            help="The path where to find libraries for which we are " "downloading the PDB files.",
+        )
+
+    def log(self, message):
+        sys.stderr.write(message + "\n")
+
+    def run(self, options: Options):
+        args = options.parsed_args
+
+        if len(sys.argv) != 5 or args.input == "" or args.libpath == "":
+            log_error("Unknown arguments passed in.")
+            return 1
+
+        self.log("Downloading PDBs from Microsoft server...")
+        self.log("Input File: " + args.input)
+        self.log("DLL Path: " + args.libpath)
+        self.log("")
+
+        if not os.path.exists(args.input):
+            log_error("Could not find " + args.input)
+            return 1
+
+        if not os.path.exists(args.libpath):
+            log_error("Could not find " + args.libpath)
+            return 1
+
+        symbol_server_url = "http://msdl.microsoft.com/download/symbols/"
+        pe = None
+        # Parse the file.
+        try:
+            pe = pefile.PE(args.input)
+        except Exception:
+            log_error("Unable to parse the input file.")
+            log_error(str(Exception))
+            return 1
+
+        for entry in pe.DIRECTORY_ENTRY_IMPORT:
```

Comment:
> It seems to me that the workflow here is:
>
>    Identify dependent DLLs
>    Download the debug info of the DLLs

True. It actually identifies all the imported (dynamic) libraries the executable depends on, and then, it tries to download for each of them a ```PDB``` from the Microsoft server.

> Why don't you download the debug info of the main PE file too?

I am not sure I get it. The ```PDB``` file for the PE file and user defined libraries compiler (and linker) produces the debug info. We are using these scripts to download debug info (the ```PDB```s) for win32 and other system libraries.

> Would it make sense to recursively download all deps of deps?

Hmm... I don't think it is a practice, since ```PDB```s could be very large, so I'd rather do that manually.

Reply:
You are right. Let's make it as simple as possible.